src/redux/missions/missions.js

```diff
@@ -0,0 +1,36 @@
+import { createAsyncThunk, createSlice } from '@reduxjs/toolkit';
+
+const initialState = { missions: [] };
+
+const URL = 'https://api.spacexdata.com/v3/missions';
+
+export const fetchMissions = createAsyncThunk('GET/fetchMissions/fulfilled', async () => {
+  const response = await fetch(URL);
+  const data = await response.json();
+  return data;
+});
+
+const missionSlice = createSlice({
+  name: 'GET',
+  initialState,
+  reducers: {
+    toogleReserve: (state, { payload }) => {
+      state.missions = state.missions.map((mission) => { // eslint-disable-line no-param-reassign
+        if (mission.mission_id === payload) {
+          return { ...mission, reserved: !mission.reserved };
+        } return mission;
```

Comment:
The toogleReserve reducer could be made more modular by extracting the logic for checking the mission ID and updating the reserved status into a separate function. This would make the reducer easier to understand and maintain.